### PR TITLE
Fixes #12380: Allow to search on machine type in search nodes request

### PR DESCRIPTION
**Pull request by Haypierre**
https://www.rudder-project.org/redmine/issues/12380

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/rudder/services/queries/LdapQueryProcessor.scala [hunk 1]
```diff
@@ -119,7 +119,7 @@ object DefaultRequestLimits extends RequestLimits(0,0,0,0)
  * for accepted nodes (it also checks that the node is registered
  * in the ou=Nodes branch)
  */
-class AccepetedNodesLDAPQueryProcessor(
+class AcceptedNodesLDAPQueryProcessor(
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/services/queries/LdapQueryProcessor.scala]: Best Change :+1:

File: rudder-core/src/main/scala/com/normation/rudder/domain/queries/CmdbQuery.scala [hunk 2]
```diff
@@ -323,6 +325,39 @@ case object MemoryComparator extends CriterionType {
...
+    }
+    comparator match {
+      case Equals => IS(v)
+      case _ => NOT(IS(v))
+    }
+  }
+
+  override def toForm(value: String, func: String => Any, attrs: (String, String)*) : Elem =
+    SHtml.select(
+      (machineTypes map (e => (e, e))).toSeq,
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/domain/queries/CmdbQuery.scala]: I know it's not done in the code you used, but, you should write this code like this
<pre>
    SHtml.select(
        (machineTypes map (e => (e, e))).toSeq
      , { if(machineTypes.contains(value)) Full(value) else Empty}
      , func
      , attrs:_*
    )
</pre>

**Comment by VinceMacBuche** [comment]
Apart the small indentation issues this is ok!  please fix them

**Comment by Haypierre** [comment]
Commit modified

**Pull request merged**